File: cpp File/15953/15953.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

int cp1_idx[101];
int cp2_idx[65];

int cp1_money[101];
int cp2_money[65];

int T, a, b, depth;

vector<int> answer;

void cal()
{
	depth = 1;

	for (int i = 1; i <= 6; i++)
	{
		for (int j = 1; j <= i; j++)
		{
			cp1_idx[depth] = i;
			depth++;
		}
	}

	depth = 1;

	for (int i = 1; i <= 5; i++)
	{
		int pow_i = pow(2, i - 1);

		for (int j = 1; j <= pow_i; j++)
		{
			cp2_idx[depth] = i;
			depth++;
		}
	}

	cp1_money[1] = 5000000;
	cp1_money[2] = 3000000;
	cp1_money[3] = 2000000;
	cp1_money[4] = 500000;
	cp1_money[5] = 300000;
	cp1_money[6] = 100000;

	cp2_money[1] = 5120000;
	cp2_money[2] = 2560000;
	cp2_money[3] = 1280000;
	cp2_money[4] = 640000;
	cp2_money[5] = 320000;
}

int main()
{
	ios_base::sync_with_stdio(0);
	cin.tie(0);
	cout.tie(0);

	cal();

	cin >> T;

	for (int i = 0; i < T; i++)
	{
		cin >> a >> b;

		int sum = cp1_money[cp1_idx[a]] + cp2_money[cp2_idx[b]];

		answer.push_back(sum);
	}

	for (int i = 0; i < answer.size(); i++)
	{
		cout << answer[i] << "\n";
	}

	return 0;
}

File: kotlin File/15953.kt
import kotlin.math.*

var depth = 0

val cp1_idx = Array<Int>(101) { 0 }
val cp2_idx = Array<Int>(65) { 0 }

val cp1_money = Array<Int>(101) { 0 }
val cp2_money = Array<Int>(65) { 0 }

fun main() {
    pre_setting()

    val ans = ArrayList<Int>()

    val T = readLine()!!.toInt()

    for (i in 0 until T) {
        val (a, b) = readLine()!!.split(' ').map { it.toInt() }

        val sum = cp1_money[cp1_idx[a]] + cp2_money[cp2_idx[b]]

        ans.add(sum)
    }

    for (i in ans) {
        println(i)
    }
}

fun pre_setting() {
    depth = 1

    for (i in 1..6) {
        for (j in 1..i) {
            cp1_idx[depth] = i
            depth++
        }
    }

    depth = 1

    for (i in 1..5) {
        val pow_i = (2.0).pow(i - 1).toInt()

        for (j in 1..pow_i) {
            cp2_idx[depth] = i
            depth++
        }
    }

    cp1_money[1] = 5000000
    cp1_money[2] = 3000000
    cp1_money[3] = 2000000
    cp1_money[4] = 500000
    cp1_money[5] = 300000
    cp1_money[6] = 100000

    cp2_money[1] = 5120000
    cp2_money[2] = 2560000
    cp2_money[3] = 1280000
    cp2_money[4] = 640000
    cp2_money[5] = 320000
}
